tests/functional/testplan/runnable/interactive/test_api.py

```diff
@@ -20,35 +21,45 @@
 class ExampleSuite(object):
     """Example test suite."""
 
+    # Note that request from client UI is processed asynchronously at server
+    # side, generally the client side will receive response at once -- just
+    # before testcases run, but that is not guaranteed. So in each tesecase
+    # a little time is added to extend execution to make functional test pass.
+
     def __init__(self, tmpfile):
         self._tmpfile = tmpfile
 
     @multitest.testcase
     def test_passes(self, env, result):
         """Testcase that passes."""
         result.true(True)
+        time.sleep(0.01)
```

Comment:
oh, the test needs to be deterministic, is it possible to set 'WAITING' before submitting the testcase to threadpool? So that 'FINISH' won't get overwritten by 'WAITING'.